### PR TITLE
add server actions to update User model

### PR DESCRIPTION
**Pull request by John-Paul-Larkin**
# ✨ Codu Pull Request 💻

## Pull Request details

Implemented 3 server actions to update the Prisma User models based on the corresponding form data.
Removed comments within the client component to enable action calls.
Renamed the action component to "_action" to bypass app folder routing.

## Summary by CodeRabbit

- **New Features**
	- Introduced new functionality for handling form submissions across multiple slides, including validation and updating of user details in the database.

### CODE REVIEW

File: app/(app)/alpha/additional-details/_actions.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    });
+    return true;
+  } catch (error) {
+    if (error instanceof z.ZodError) {
+      console.error("Validation error:", error.errors);
+    } else {
+      console.error("Error updating the User model:", error);
+    }
+    return false;
+  }
```
[the remark this answers is not in the document]
Comment (John-Paul-Larkin) [reply app/(app)/alpha/additional-details/_actions.ts:46]: Not sure about this one @NiallJoeMaher 

Redirect is not used server side anywhere else in the code base, but according to the next docs :

>You can call redirect in Server Components, Route Handlers, and **Server Actions**.

[https://nextjs.org/docs/app/building-your-application/routing/redirecting#redirect-function](docs)

[the remark this answers is not in the document]
Comment (NiallJoeMaher) [reply app/(app)/alpha/additional-details/_actions.ts:46]: I think you are right here JP! So let's test it out this way.

[the remark this answers is not in the document]
Comment (John-Paul-Larkin) [reply app/(app)/alpha/additional-details/_actions.ts:46]: > I think you are right here JP! So let's test it out this way.

Maybe I should include a return statement after the redirects to ensure the code halts execution at that point?

File: app/(app)/alpha/additional-details/_actions.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply app/(app)/alpha/additional-details/_actions.ts:75]: Question on the /alpha

Is this how you mark stuff thats in testing? As opposed to doing process.env.ALPHA ?

[the remark this answers is not in the document]
Comment (John-Paul-Larkin) [reply app/(app)/alpha/additional-details/_actions.ts:75]: > Question on the /alpha
> 
> Is this how you mark stuff thats in testing? As opposed to doing process.env.ALPHA ?

Hey John.
Yes, that is my understanding of it. 
As far as I can tell the alpha folder is only available in development, not on the www.dev1.codu.co version, or production.(obviously) Im sure the boss will be along soon to clarify. :smile:

**Review by NiallJoeMaher — APPROVED** [review]

File: app/(app)/alpha/additional-details/_actions.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply app/(app)/alpha/additional-details/_actions.ts:75]: Thanks @John-Paul-Larkin  makes sense. @NiallJoeMaher care to weigh in?

**Pull request merged**